consensus/wal.go

```diff
@@ -72,6 +75,9 @@ type baseWAL struct {
 	group *auto.Group
 
 	enc *WALEncoder
+
+	syncTicker *time.Ticker
+	testChan   chan string
```

Comment:
I'd prefer not to leak test concerns into implementation. If you want to test processSyncTicks function, you can create processSyncTick func and call it **OR** lower the interval in tests and assert the result after timeout.

```
func processSyncTicks() {
  for {
		select {
		case <-wal.syncTicker.C:
                  processSyncTick()
..
}
  }
}
```

```
wal := NewWAL(syncInterval: 10ms)
wal.Start()

time.Sleep(11ms)

assert.Synced(t, wal)
```

Reply:
This is a temporary measure until I implement that `testBus` we were talking about. I'd prefer not to add more `time.Sleep()` calls to the tests 😁

Can I just add a `TODO` note here for now rather and then remove this once I've got the `testBus` implementation started (in a separate PR)?

Reply:
Let's run the `testBus` idea against @ebuchman first before moving on with it.

Reply:
👍